File: java/src/main/java/mx/iteso/adapter/WordAdapter.java
package mx.iteso.adapter;

/** Word to GoogleDoc.*/
public class WordAdapter implements IGoogleDoc {

    /** Word.*/
    private WordDocument word;

    /** License.*/
    private MSLicense ms = new MSLicense("MS License");

    /** Format.*/
    private Format format;

    /** Font.*/
    private Font font;

    /** Style.*/
    private String style;

    /** @param wordDoc . */
    public WordAdapter(final WordDocument wordDoc) {
        this.word = wordDoc;
        font = new Font("Arial");
        style = "bold";
        format = new Format(font, style);
    }

    /** @return Font.*/
    public Font getFont() {
        return this.font;
    }

    /** @return Style.*/
    public Object getStyle() {
        return this.style;
    }

    /** @return Format.*/
    public Object gatFormat() {
        return this.format;
    }

    /** @return Background.*/
    public BackgroundImage getBackground() {
        return new BackgroundImage(word.getBackground());
    }

    /** @param sharingPermissions .*/
    public void setSharingPermissions(final int sharingPermissions) {
        word.setMSOfficeVersion(sharingPermissions);
        word.getLicense();
        word.restrictEditIfLicenseIsInvalid(ms);
    }
}

File: csharp/adapter-practice/WordAdapter.cs
using System;


namespace adapter_practice
{
    public class WordAdapter : IGoogleDoc
    {

        private readonly IWordDocument _wordDocument;

        private readonly MsLicense _ms = new MsLicense("MS License");

        private readonly Format _format;


        public WordAdapter(IWordDocument wordDoc)
        {
            this._wordDocument = wordDoc;
            Font font = new Font("Verdana");
            String style = "style";
            _format = new Format(font, style);
        }

        public Font GetFont()
        {
            return _format.ReturnFont();
        }

        public Object GetStyle()
        {
            return _format.GetStyle();
        }

        public BackgroundImg GetBackground()
        {
            return new BackgroundImg(_wordDocument.GetBackground());
        }

        public void SetSharingPermissions(int sharingPermissions)
        {
            _wordDocument.SetMOfficeVersion(sharingPermissions);
            _wordDocument.GetLicense();
            _wordDocument.RestrictEditIfLicenseIsInvalid(_ms);
        }
    }
}
